### ISSUE TITLE
walking diagonally through large doors - impossible?

### ISSUE THREAD

**Issue by lynxlynxlynx** [ja2-stracciatella/ja2-stracciatella#83]
Originally reported by: **JAsmine-ja2 (Bitbucket: [JAsmine-ja2](http://bitbucket.org/JAsmine-ja2), GitHub: [JAsmine-ja2](http://github.com/JAsmine-ja2))**

---

This is a minor issue, that was already present in vanilla JA2.
![issue with doors](http://img.xrmb2.net/images/679331.jpeg)
It is not possible to walk through large doors diagonally.
The merc will always turn two times instead, if you try to do it.
You can see the movement path in the screenie.

This is true for all doors and both angles.

---
- Bitbucket: https://bitbucket.org/gennady/ja2-stracciatella/issue/83

**Comment by derSteFfi** [comment]
because i just came across it: this is rejected in Build/Tactical/PathAI.cc, currently around line 1075:
```C
				else if ( IS_TRAVELCOST_DOOR( nextCost ) )
				{

					// don't let anyone path diagonally through doors!
					if (iCnt & 1)
					{
						goto NEXTDIR;
					}
```

**Comment by lynxlynxlynx** [comment]
I guess that is in for normal doors, where it makes sense. For larger we should skip the check at least in the internally angled diagonals. However, I have a nagging feeling that everything is cell based here and there is no concept of a wide door. If that's true, we have to be careful not to break single doors that are side by side.

**Comment by RinneT** [comment]
I deleted the check @derSteFfi posted just to see what happens.
Walking through the double wide door from the screenshot above works correctly, so does walking through a single door.
The same behaviour as shown above does happen on two side by side doors as in cambria hospital.
It seems the two single doors are handled differently.

Essentially the code is useless.

I attached two savegames for testing.
[Issue83_Doors.zip](https://github.com/ja2-stracciatella/ja2-stracciatella/files/614754/Issue83_Doors.zip)
